mirage/utils.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable max-statements-per-line */
+import {Response} from 'ember-cli-mirage';
+
+export function paginatedResponse(modelName) {
+    return function (schema, request) {
+        let page = +request.queryParams.page || 1;
+        let limit = request.queryParams.limit || 15;
+        let pages, next, prev, models;
+
+        let allModels = this.serialize(schema[modelName].all())[modelName];
+
+        if (limit === 'all') {
+            pages = 1;
+        } else {
+            limit = +limit;
+
+            let start = (page - 1) * limit;
+            let end = start + limit;
+
+            pages = Math.ceil(allModels.length / limit);
+            models = allModels.slice(start, end);
+
+            if (start > 0) {
+                prev = page - 1;
+            }
+
+            if (end < allModels.length) {
+                next = page + 1;
+            }
+        }
+
+        return {
+            meta: {
+                pagination: {
+                    page,
+                    limit,
+                    pages,
+                    total: allModels.length,
+                    next: next || null,
+                    prev: prev || null
+                }
+            },
+            [modelName]: models || allModels
+        };
+    };
+}
+
+export function maintenanceResponse() {
+    return new Response(503, {}, {
+        errors: [{
+            errorType: 'Maintenance'
+        }]
+    });
+}
+
+export function versionMismatchResponse() {
+    return new Response(400, {}, {
+        errors: [{
+            errorType: 'VersionMismatchError'
+        }]
+    });
+}
+
+// minimalist deparam implementation
+export function deparam(string) {
```

Comment:
Yeah, `jquery-deparam` is only pulled in for testing at the moment. I think it may be better to include it on all builds as it's a pretty small dependency and there are a few places we do some not so robust checks for url params. Hopefully Ember's router will expose it's param handling at some point.

I think the main issue with the minimalist implementation here is that it doesn't cover the `?foo[]=bar` type syntax - I'm not sure we're using that anywhere at the moment but it's quite likely to crop up with post filtering.